assembly/src/serde/mod.rs

```diff
@@ -69,56 +60,53 @@ impl ByteWriter {
 
 /// Converts `self` into bytes and writes them to the provided `ByteWriter` struct
 pub trait Serializable: Sized {
-    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError>;
+    fn write_into(&self, target: &mut ByteWriter);
 
-    fn to_bytes(&self) -> Result<Vec<u8>, SerializationError> {
+    fn to_bytes(&self) -> Vec<u8> {
         let mut target = ByteWriter::default();
-        self.write_into(&mut target)?;
-        Ok(target.into_bytes())
+        self.write_into(&mut target);
+        target.into_bytes()
     }
 }
 
 impl Serializable for () {
-    fn write_into(&self, _target: &mut ByteWriter) -> Result<(), SerializationError> {
-        Ok(())
-    }
+    fn write_into(&self, _target: &mut ByteWriter) {}
 }
 
 impl Serializable for &'_ str {
-    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
-        target.write_str(self)
+    fn write_into(&self, target: &mut ByteWriter) {
+        target.write_str(self);
     }
 }
 
 impl Serializable for String {
-    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
-        target.write_str(self)
+    fn write_into(&self, target: &mut ByteWriter) {
+        target.write_str(self);
     }
 }
```

Comment:
I would remove these implementations. And also would remove implementations for serializing `Vec<T>` below.